infra/src/main/scala/ch/epfl/scala/services/github/GithubImplementation.scala

```diff
@@ -0,0 +1,236 @@
+package ch.epfl.scala.services.github
+
+import java.time.Instant
+
+import scala.concurrent.ExecutionContextExecutor
+import scala.concurrent.Future
+import scala.concurrent.Promise
+import scala.concurrent.duration.DurationInt
+import scala.util.Failure
+import scala.util.Success
+import scala.util.Try
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.HttpHeader
+import akka.http.scaladsl.model.HttpRequest
+import akka.http.scaladsl.model.HttpResponse
+import akka.http.scaladsl.model.ResponseEntity
+import akka.http.scaladsl.model.StatusCodes
+import akka.http.scaladsl.model.headers.OAuth2BearerToken
+import akka.http.scaladsl.model.headers.RawHeader
+import akka.http.scaladsl.settings.ConnectionPoolSettings
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.OverflowStrategy
+import akka.stream.QueueOfferResult
+import akka.stream.scaladsl.Keep
+import akka.stream.scaladsl.Sink
+import akka.stream.scaladsl.Source
+import akka.stream.scaladsl.SourceQueueWithComplete
+import akka.util.ByteString
+import ch.epfl.scala.index.model.misc.GithubInfo
+import ch.epfl.scala.index.model.misc.GithubRepo
+import ch.epfl.scala.index.model.misc.Url
+import ch.epfl.scala.services.GithubService
+import ch.epfl.scala.services.github.GithubModel.Contributor
+import ch.epfl.scala.utils.ScalaExtensions.TraversableOnceFutureExtension
+import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport._
+
+class GithubImplementation(githubConfig: GithubConfig)(implicit system: ActorSystem) extends GithubService {
```

Comment:
Maybe `AkkaGithubClient` or just `GithubClient`?

Reply:
I don't really like the name `Implementation`...

Reply:
Agree